### PR TITLE
feat(tauri): unit tests, benchmarks, docs (phase 5)

### PR DESCRIPTION
**Pull request by goosewobbler**
Phase 5 of the Tauri-Electron alignment plan. Stacks on top of #149 (which lands the renderer architecture port + plugin restructure). Targets `next/tauri-alignment` so it can be combined with the other phases before merging to `main`.

## Summary

- **Unit tests** — ported / adapted from `packages/electron/test/` for every duplicated TS module in `packages/tauri/src/`.
- **Benchmarks** — `packages/tauri/benchmarks/` mirroring the Electron suite, plus a new dispatch-latency benchmark exercising the bridge end-to-end against a mocked invoke/listen transport.
- **READMEs** — full rewrite of `packages/tauri/README.md` (with a 1.x → 2.x migration guide) and `packages/tauri-plugin/README.md` (Rust API, command list, wire payloads, `StateManager` trait).

No source changes — every `src/` directory, the existing integration suite at `packages/tauri/test/index.spec.tsx`, the Electron package, and the Tauri E2E app are untouched.

## Coverage delta

| Suite | Before | After | Delta |
| --- | --- | --- | --- |
| Test files | 1 | 8 | +7 |
| Test cases | 23 | 232 | +209 |

New unit specs (with the per-spec adjustments to match the Tauri implementation):

| Spec | Adaptation from Electron |
| --- | --- |
| `test/deltas/DeltaMerger.spec.ts` | byte-identical port |
| `test/batching/ActionBatcher.spec.ts` | byte-identical port |
| `test/serialization.spec.ts` | byte-identical port |
| `test/errors/index.spec.ts` | `IpcCommunicationError` → `TauriCommandError` (with `command` / `sourceLabel`); `ActionProcessingError` accepts the `'tauri'` adapter; `SubscriptionError` carries `sourceLabel` |
| `test/renderer/actionValidator.spec.ts` | `setActionValidatorStateProvider(provider)` replaces the `window.zubridge.getState` mock |
| `test/renderer/subscriptionValidator.spec.ts` | `setSubscriptionFetcher(fetcher)` replaces the `window.__zubridge_subscriptionValidator` API mock; tests the real Tauri matching logic (default-all on empty subs, sibling-prefix false-positive guard, cache TTL) |
| `test/renderer/rendererThunkProcessor.spec.ts` | webview `windowLabel: string` replaces numeric `windowId`; the actionSender's resolution auto-completes the action (no separate ack); `window.zubridge` fallback dispatch path is dropped (Tauri renderers do not get an injected global) |

The new specs target edge cases the existing 23-test integration suite at `test/index.spec.tsx` cannot easily reach in isolation.

## Benchmarks

`packages/tauri/benchmarks/` (run with `pnpm --filter @zubridge/tauri bench`):

- `batching.bench.ts` — verbatim port: IPC-call reduction baseline, batcher throughput at 10/100 actions, priority-flush overhead.
- `delta.bench.ts` — renderer-only port (the Tauri `DeltaCalculator` lives in Rust): single-key, deep-path, removed-keys, full-vs-delta, and 20-key stress merges; payload-size comparison.
- `dispatch.bench.ts` — new: `useZubridgeDispatch` latency through `BridgeClient` against a fully-mocked `invoke` + `listen` transport. String / object / deeply-nested-payload single dispatches plus 10/50-action sequential bursts.

Files use the `.bench.ts` extension to match vitest's default benchmark pattern, so no `vitest.config.ts` changes are required. The `bench: vitest bench --run` script in `packages/tauri/package.json` mirrors the Electron package.

## READMEs

- **`packages/tauri/README.md`** — full new API reference (`initializeBridge`, `useZubridgeStore`, `useZubridgeDispatch`, thunks, `subscribe` / `unsubscribe` / `getWindowSubscriptions`, `dispatch.batch`, validators, `TauriCommandError`), wire-protocol table covering plugin and direct command paths, state-update event payload shape, and a 1.x → 2.x migration guide (action wire shape, `updateState` removal, subscription API, `cleanup` → `cleanupZubridge`, error rename, plugin version bump).
- **`packages/tauri-plugin/README.md`** — Rust command reference with args/results, `StateManager` trait contract, plugin entry points (`plugin` / `plugin_default` / `init`), `StateUpdatePayload` and `StateDelta` shapes, `Error` variants, and the security guarantee that webview labels are derived from `tauri::Window<R>::label()` rather than client args.

## Validation

- `pnpm --filter @zubridge/tauri test:unit` — 232/232 ✓
- `pnpm --filter @zubridge/tauri typecheck` — clean
- `pnpm --filter @zubridge/tauri bench --run` — every benchmark suite completes without error

## Commit style

Small logical commits per file group:

- `test(tauri): port DeltaMerger unit tests from electron`
- `test(tauri): port ActionBatcher unit tests from electron`
- `test(tauri): port error registry tests with TauriCommandError`
- `test(tauri): port actionValidator with provider injection`
- `test(tauri): port subscriptionValidator with fetcher injection`
- `test(tauri): port rendererThunkProcessor with webview-label semantics`
- `test(tauri): port sanitizeState serialization tests from electron`
- `bench(tauri): port + adapt vitest benchmarks for the new TS package`
- `docs(tauri): rewrite README for the 2.x renderer API`
- `docs(tauri-plugin): rewrite README for the 0.2.x Rust API`

https://claude.ai/code/session_0163nGnsThc7CrDLcEUkXXU9

### CODE REVIEW

File: packages/tauri-plugin/src/desktop.rs [hunk 1]
```diff
@@ -1,83 +1,404 @@
...
+    }
+
+    /// Register a state manager at runtime (used when the plugin is initialised
+    /// without one).
+    pub fn register_state_manager<S: StateManager>(
+        &self,
+        state_manager: S,
+    ) -> crate::Result<()> {
+        let handle = state_manager::new_handle(state_manager);
+        self.app.manage(handle);
```
[the remark this answers is not in the document]
Comment (goosewobbler) [reply packages/tauri-plugin/src/desktop.rs:508]: Addressed in #149

File: packages/tauri-plugin/src/desktop.rs [hunk 2]
```diff
@@ -1,83 +1,404 @@
...
+        let mut tracker = self
+            .update_tracker
+            .write()
+            .map_err(|e| crate::Error::StateError(e.to_string()))?;
+        tracker.ack(source_label, update_id);
+        Ok(())
+    }
+
+    /// Subscribe a webview to a set of top-level state keys.
+    pub fn subscribe(&self, source_label: &str, keys: &[String]) -> crate::Result<Vec<String>> {
```
[the remark this answers is not in the document]
Comment (goosewobbler) [reply packages/tauri-plugin/src/desktop.rs:335]: Addressed in #149

File: packages/tauri/src/types/tauri.ts [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+  /** Request batching configuration. Pass `false` to disable batching entirely. */
+  batching?: BatchingOptions | false;
+}
+
+export interface BatchingOptions {
+  /** Max actions per batch before forcing a flush. Default 100. */
+  maxBatchSize?: number;
+  /** Flush window in milliseconds. Default 5ms. */
+  windowMs?: number;
+}
```
[the remark this answers is not in the document]
Comment (goosewobbler) [reply packages/tauri/src/types/tauri.ts:58]: Fixed in 39d816f. JSDoc now matches `BATCHING_DEFAULTS` (`maxBatchSize: 50`, `windowMs: 16`) and the README has a dedicated defaults table plus a note about the `batching: false` opt-out so consumers don't need to chase the values across files.

Strictly this lives in `packages/tauri/src/`, which Phase 5 wasn't supposed to touch — but it's a 3-line comment-only correction with no behaviour / type / wire-protocol impact, and leaving it stale through Phase 5 would actively mislead anyone reading the new README's `BatchingOptions` reference. Routed it here rather than back to #149.

---
_Generated by [Claude Code](https://claude.ai/code)_
